### PR TITLE
Fix default requests including the API resource path

### PR DESCRIPTION
**Pull request by Shigawire**
Authentication logic is provided by `doorkeeper` at root-level of the application (`/spree_oauth/token`).

In this project, however, all calls are prefixed with `api/v2/storefront`, which obviously interferes with doorkeeper's logic and as such, no authentication calls are possible with the library:

```
Started POST "/api/v2/storefront/spree_oauth/token" for 127.0.0.1 at 2019-03-23 19:28:30 +0100
Started GET "/api/v1/storefront/spree_oauth/token.json" for 127.0.0.1 at 2019-03-23 19:28:30 +0100
Started GET "/api/404" for 127.0.0.1 at 2019-03-23 19:28:30 +0100
```

This PR removes the forced URL prefix and attaches this to all calls in the `routes.ts` where necessary.
Maybe there is a finer or more DRY solution. As my background is rather on Rails, this is what I could come up with.

### CODE REVIEW

File: src/routes.ts [hunk 1]
```diff
@@ -1,30 +1,33 @@
+
```
Comment (tniezg) [inline src/routes.ts]: Let's remove the extra line at the top.

File: src/routes.ts [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
+const baseUrl = `api/v2/storefront`;
```
Comment (tniezg) [inline src/routes.ts]: Can we use `storefrontUrl` instead of `baseUrl`, please? Will be more descriptive.

**Comment by tniezg** [comment]
Good catch! @Shigawire

File: src/routes.ts (hunk 2 above)
Comment (Shigawire) [reply src/routes.ts]: Actually, storefrontPath would be even more suitable, but I changed it to `storefrontUrl` now.

File: src/routes.ts [hunk 3]
```diff
@@ -1,30 +1,32 @@
+const storefrontUrl = `api/v2/storefront`;
```
Comment (tniezg) [inline src/routes.ts]:
```suggestion
const storefrontUrl = `api/v2/storefront`
```

File: src/routes.ts (hunk 2 above)
Comment (tniezg) [reply src/routes.ts]: You're right, let's use `storefrontPath`.

**Comment by tniezg** [comment]
@Shigawire Can you squash changes into the first commit? No need to have a separate commit with "QA" fixes :)

**Comment by Shigawire** [comment]
@tniezg Will update. Squashing can only be done upon merging. Is that ok for you? Otherwise I would force push to the first commit, but IMO, this is bad style.

File: src/Http.ts [hunk 4]
```diff
@@ -18,7 +18,7 @@ export default class Http {
     this.host = process.env.SPREE_HOST || 'http://localhost:3000/'
```
Comment (Shigawire) [inline src/Http.ts]: @tniezg if `process.env.SPREE_HOST` is not followed by a `/` then the absolute `storefrontPath` will not work. Thus, I would change the fallback to `http://localhost:3000` and append a `/` to the storefrontPath.

Comment (damianlegawiec) [reply src/Http.ts]: @Shigawire if you could change it would be awesome - thank you!

Comment (tniezg) [reply src/Http.ts]: @Shigawire It works in both cases for me. What host did you try exactly? Was it `http://localhost:3000`?

Comment (Shigawire) [reply src/Http.ts]: To be honest, I didn't try. But logically, using a host without trailing slash as `baseUrl` and then appending the `storefrontPath` without a leading `/` should be error-prone, not?

Comment (tniezg) [reply src/Http.ts]: Axios handles the case with missing `/`, so we can leave it as is.

Comment (Shigawire) [reply src/Http.ts]: Fair enough then.

**Pull request merged**